package.xml

```diff
@@ -10,8 +10,8 @@
 
   <license>Apache License 2.0</license>
 
-  <exec_depend condition="$ROS_PYTHON_VERSION == 2">python-mock</exec_depend>
-  <exec_depend condition="$ROS_PYTHON_VERSION == 3">python3-mock</exec_depend>
+  <exec_depend>python3-importlib-metadata</exec_depend>
```

Comment:
So, this dependency is not available on Xenial or Bionic, which I need it to be to make debian releases:

https://github.com/ros/rosdistro/blob/c71e47e3cc0e5b7876148e1d934a38447c8b6429/rosdep/python.yaml#L6543-L6548

(also confirmed with my Bionic VM)

So I don't think we can take this pull request. We'll have to make separate releases for before and after bionic I guess? @clalancette I'm gonna move forward with a Python3 only release without this pull request until we can resolve this point.

Reply:
Yeah, that's correct.  importlib-metadata is only available as a Debian package with Ubuntu 20.04 and later.

Backing up a bit, I don't understand exactly how this package is released.  I thought it was released as a pip package or a ROS package, both cases in which case the package.xml in this PR would work.  Is there another scenario in which this is released?  Could you explain a little more about it?

Reply:
> Backing up a bit, I don't understand exactly how this package is released. I thought it was released as a pip package or a ROS package, both cases in which case the package.xml in this PR would work. Is there another scenario in which this is released? Could you explain a little more about it?

It's actually released on pip and as a deb via stdeb using our `ros_release_python` toolchain, and in ROS 2 specifically it is bloomed.

In the pip case, everything is fine.

In the stdeb case (i.e. `python3-osrf-pycommon`  via packages.ros.org) that's a problem because the `python3-osrf-pycommon` deb that is produced has no deb to depend on in Xenial or Bionic.

In the ROS 2 case, similarly, since the rosdep rule points to a pip install, it cannot be built/released via our buildfarm.

Is that what you're asking?

Reply:
> was released as a pip package or a ROS package, both cases in which case the package.xml in this PR would work.

I don't think it works in either of those cases, unless you're counting a source build, in which case rosdep would install from pip.

I suppose we could limit the ROS 2 release to things that don't use Bionic, but that's not sufficient, and it would mean effectively never releasing for Xenial or Bionic again.